### PR TITLE
add context switchers for org IAM

### PR DESCRIPTION
**Pull request by micheleRP**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2441

This adds context switchers for AWS/GCP to make the content more readable and also as a first step toward using context switchers for cloud providers in other Cloud docs. It also fixes minor issues in other Cloud security pages, including changing links into self-hosted pages into glossary entries. 

Review deadline: Monday, May 20

## Page previews
- https://deploy-preview-499--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/cloud/security/authorization/cloud-iam-policies/
- https://deploy-preview-499--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/cloud/security/authorization/cloud-iam-policies-gcp/

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [X] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/deploy/pages/deployment-option/cloud/security/cloud-availability.adoc [hunk 1]
```diff
@@ -3,10 +3,10 @@
 :page-cloud: true
 
 Redpanda Cloud supports the deployment of Redpanda clusters in single or multiple
-availability zones (AZs), spanning at most three AZs. Nodes are evenly distributed
+glossterm:availability zone (AZ)[,AZs], spanning at most three AZs. Brokers are evenly distributed
```
Comment (JakeSCahill) [inline modules/deploy/pages/deployment-option/cloud/security/cloud-availability.adoc]:
```suggestion
glossterm:availability zones[,AZs], spanning at most three AZs. Brokers are evenly distributed
```

Comment (JakeSCahill) [reply modules/deploy/pages/deployment-option/cloud/security/cloud-availability.adoc]: The term name is availability zones: https://raw.githubusercontent.com/redpanda-data/docs/shared/modules/terms/partials/az.adoc

It should be singular really, but to get this to work, you must match the defined term name.

**Review by JakeSCahill — APPROVED** [review]
Looks so good. Love the split for different audiences.

I left a comment to fix the glossterm.

**Review by Deflaimun — APPROVED** [review]
Looks nice

**Pull request merged**